src/vs/platform/extensionManagement/node/extensionsScanner.ts

```diff
@@ -59,39 +47,29 @@ export class ExtensionsScanner extends Disposable {
 	}
 
 	async scanExtensions(type: ExtensionType | null): Promise<ILocalExtension[]> {
-		const promises: Promise<ILocalExtension[]>[] = [];
-
-		if (type === null || type === ExtensionType.System) {
-			promises.push(this.scanDefaultSystemExtensions());
-			promises.push(this.environmentService.isBuilt ? Promise.resolve([]) : this.scanDevSystemExtensions());
-		} else {
-			promises.push(Promise.resolve([]));
-			promises.push(Promise.resolve([]));
-		}
-		promises.push(this.scanUserExtensions(false));
-
-		try {
-			const [defaultSystemExtensions, devSystemExtensions, userExtensions] = await Promise.all(promises);
-			const result = this.dedupExtensions([...defaultSystemExtensions, ...devSystemExtensions, ...userExtensions], await this.targetPlatform);
-			return type !== null ? result.filter(r => r.type === type) : result;
-		} catch (error) {
-			throw this.joinErrors(error);
+		const scannedOptions: ScanOptions = { includeInvalid: true };
+		let scannedExtensions: IScannedExtension[] = [];
+		if (type === null) {
+			scannedExtensions.push(...await this.extensionsScannerService.scanAllExtensions(scannedOptions));
+		} else if (type === ExtensionType.System) {
+			scannedExtensions.push(...await this.extensionsScannerService.scanSystemExtensions(scannedOptions));
+			scannedExtensions.push(...await this.extensionsScannerService.scanUserExtensions(scannedOptions));
```

Comment:
Should this line be here?

Reply:
Yeah it should be. Because updated system extensions are located in user extensions folder. At the end, I filter for system extensions and return the result.

I can simplify it by merging first two if branches